lms/djangoapps/courseware/views/index.py

```diff
@@ -152,34 +148,25 @@ def get(self, request, course_id, chapter=None, section=None, position=None):
 
                 self.view = STUDENT_VIEW
 
-                # Do the enrollment check if enable_unenrolled_access is not enabled.
                 self.course = get_course_with_access(
```

Comment:
So the logic here in `get` is one thing we need to check carefully.  It's very different, but _should_ be completely equivalent.